core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java

```diff
@@ -91,6 +95,31 @@
     @Inject
     private UsageReporter usageReporter;
 
+    public void vmDiagnosticsReport(String resourceCrn, DiagnosticParameters parameters) {
+        vmDiagnosticsReport(resourceCrn, parameters, null, null);
+    }
+
+    public void vmDiagnosticsReport(String resourceCrn, DiagnosticParameters parameters, UsageProto.CDPVMDiagnosticsFailureType.Value failureType,
+            Exception exception) {
+        UsageProto.CDPVMDiagnosticsEvent.Builder eventBuilder = UsageProto.CDPVMDiagnosticsEvent.newBuilder();
+        if (failureType != null) {
+            eventBuilder.setFailureType(failureType);
+        }
+        if (exception != null) {
+            eventBuilder.setFailureMessage(StringUtils.left(exception.getMessage(), ERROR_MESSAGE_MAX_LENGTH));
+            eventBuilder.setResult(UsageProto.CDPVMDiagnosticsResult.Value.FAILED);
+        } else {
+            eventBuilder.setResult(UsageProto.CDPVMDiagnosticsResult.Value.SUCCESSFUL);
+        }
+        eventBuilder.setUuid(parameters.getUuid());
+        eventBuilder.setDescription(parameters.getDescription());
+        eventBuilder.setAccountId(parameters.getAccountId());
+        eventBuilder.setInputParameters(parameters.toMap().toString());
+        eventBuilder.setResourceCrn(resourceCrn);
```

Comment:
It would be safer to use defaultIfEmpty() or similar when setting these fields, because if the getters in "parameters" return null, then the builder will throw NPE